core/tests/puppeteer-acceptance-tests/user-utilities/voiceover-admin-utils.ts

```diff
@@ -0,0 +1,150 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility functions for voiceover admin page
+ */
+
+import {BaseUser} from '../puppeteer-testing-utilities/puppeteer-utils';
+import testConstants from '../puppeteer-testing-utilities/test-constants';
+
+const creatorDashboardPage = testConstants.URLs.CreatorDashboard;
+
+const createExplorationButton = 'button.e2e-test-create-new-exploration-button';
+const dismissWelcomeModalSelector = 'button.e2e-test-dismiss-welcome-modal';
+const textStateEditSelector = 'div.e2e-test-state-edit-content';
+const richTextAreaField = 'div.e2e-test-rte';
+const saveContentButton = 'button.e2e-test-save-state-content';
+const addInteractionButton = 'button.e2e-test-open-add-interaction-modal';
+const interactionEndExplorationInputButton =
+  'div.e2e-test-interaction-tile-EndExploration';
+const saveInteractionButton = 'button.e2e-test-save-interaction';
+const saveChangesButton = 'button.e2e-test-save-changes';
+const saveDraftButton = 'button.e2e-test-save-draft-button';
+
+const publishExplorationButton = 'button.e2e-test-publish-exploration';
+const explorationTitleInput = 'input.e2e-test-exploration-title-input-modal';
+const explorationGoalInput = 'input.e2e-test-exploration-objective-input-modal';
+const explorationCategoryDropdown =
+  'mat-form-field.e2e-test-exploration-category-metadata-modal';
+const saveExplorationChangesButton = 'button.e2e-test-confirm-pre-publication';
+const explorationConfirmPublishButton = 'button.e2e-test-confirm-publish';
+const closeShareModalButton = 'button.e2e-test-share-publish-close';
+
+const explorationSettingsTab = '.e2e-test-settings-tab';
+const editVoiceoverArtistButton = 'span.e2e-test-edit-voice-artist-roles';
+const voiceArtistEditSelector = 'input.e2e-test-new-voice-artist-username';
+const saveVoiceoverArtistEditButton =
+  'button.e2e-test-add-voice-artist-role-button';
+
+const errorToastMessage = 'div.e2e-test-toast-warning-message';
+const closeToastMessageButton = 'button.e2e-test-close-toast-warning';
+
+const updatedVoiceoverArtist = 'div.e2e-test-voice-artist-voiceoverartist';
+
+export class VoiceoverAdmin extends BaseUser {
+  /**
+   * Function to navigate to creator dashboard page
+   */
+  async navigateToCreatorDashboardPage(): Promise<void> {
+    await this.page.goto(creatorDashboardPage);
+  }
+
+  /**
+   * Function to navigate to exploration settings tab
+   */
+  async navigateToExplorationSettingsTab(): Promise<void> {
+    await this.page.waitForTimeout(500);
+    await this.clickOn(explorationSettingsTab);
+    await this.page.waitForTimeout(500);
```

Comment:
Will removing the `waitForTimeout` statements cause the test to timeout? If so, then I think it's worth investigating why that is the case. Generally, you should consider the other `waitFor...` methods before resorting to `waitForTimeout`. Regardless, the end goal is to have all of these statements removed.

Ditto for all other `waitForTimeout` instances.

Reply:
I see, Updated the test take a look